agent/rpc/peering/service.go

```diff
@@ -490,9 +491,14 @@ func (s *Service) StreamResources(stream pbpeering.PeeringService_StreamResource
 	if !s.Backend.IsLeader() {
 		// we are not the leader so we will hang up on the dialer
 
-		// TODO(peering): in the future we want to indicate the address of the leader server as a message to the dialer (best effort, non blocking)
 		s.logger.Error("cannot establish a peering stream on a follower node")
-		return grpcstatus.Error(codes.FailedPrecondition, "cannot establish a peering stream on a follower node")
+		var merr *multierror.Error
```

Comment:
not really sure if we want to use `merr` here.

we could just best effort log `sendLeaderAddrMsg` and drop the error on the floor; that kinda feels wrong.

what I don't like is that now the return error is not a `grpcstatus.Error` 🤔 

wdyt? remove?

Reply:
I think it might be better to only log the `sendLeaderAddrMsg` error and not try to reply with it with multierror. This is purely an opportunistic kind of attempt to send some data, so if it doesn't work it's not like the other peer server can do anything with this error since it's going to get the OTHER error at the same time and do the same thing regardless.

Reply:
we changed approach so now there's no more `sendLeaderAddrMsg`